src/main/java/org/dice_research/opal/civet/metrics/DateFormatMetric.java

```diff
@@ -38,76 +31,79 @@
  */
 public class DateFormatMetric implements Metric {
 
-	private static final Logger LOGGER = LogManager.getLogger();
 	private static final String DESCRIPTION = "If the date format is in the correct format then give 5 stars else 0 star"
 			+ "Finally, an average of the following 4 cases has to be computed for the finalscore: dataset: issued + modified, distributions: issued + modified";
 
-	private static List<String> correctFormat = new ArrayList<String>();
-	static {
-		correctFormat.add("YYYY");
-		correctFormat.add("YYYY-MM");
-		correctFormat.add("YYYY-MM-DD");
-		correctFormat.add("YYYY-MM-DDThh:mmTZD");
-		correctFormat.add("YYYY-MM-DDThh:mm:ssTZD");
-		correctFormat.add("YYYY-MM-DDThh:mm:ss.sTZD");
-	}
-
-	public static int checkDateFormat(String dct_issued) {
-		for (String string : correctFormat) {
-			SimpleDateFormat format = new SimpleDateFormat(string);
-			try {
-				format.parse(dct_issued.trim());
-				return 5;
-
-			} catch (ParseException pe) {
-				continue;
-			}
+	public static int checkDateFormat(String issued) {
+		/*
+		 * Following are the correct date formats according to "w3.org". Here, we are
+		 * checking different date formats using regular expression.
+		 * 
+		 */
+		if (issued.matches("\\d{4}-\\d{2}-\\d{2}")) {
+			// "YYYY-MM-DD"
+			return 5;
+		}
+		if (issued.matches("\\d{4}")) {
+			// "YYYY"
+			return 5;
+		}
+		if (issued.matches("\\d{4}-\\d{2}")) {
+			// "YYYY-MM"
+			return 5;
+		}
+		if (issued.matches("\\d{4}-\\d{2}-\\d{2}[T]\\d{2}:\\d{2}:\\d{2}[+]\\d{2}:\\d{2}")) {
+			// "YYYY-MM-DDThh:mm:ssTZD"
+			return 5;
+		}
+		if (issued.matches("\\d{4}-\\d{2}-\\d{2}[T]\\d{2}:\\d{2}[+]\\d{2}:\\d{2}")) {
+			// "YYYY-MM-DDThh:mmTZD"
+			return 5;
+		}
+		if (issued.matches("\\d{4}-\\d{2}-\\d{2}[T]\\d{2}:\\d{2}:\\d{2}[.]\\d+[+]\\d{2}:\\d{2}")) {
+			// "YYYY-MM-DDThh:mm:ss.sTZD"
+			return 5;
 		}
 		return 0;
 	}
 
 	@Override
 	public Integer compute(Model model, String datasetUri) throws Exception {
 		int result = 0;
-		LOGGER.info("Processing dataset " + datasetUri);
+		int countIssued = 0;
+		int countModified = 0;
 		Resource dataSet = model.createResource(datasetUri);
 		if (dataSet.hasProperty(DCTerms.issued)
 				&& !(dataSet.getProperty(DCTerms.issued).getObject().toString().isEmpty())) {
```

Comment:
This branch condition will not check the following case.
dct:issued         "";  or when 
dct:modified         "";

And the dataset will get full 5-star ratings. So how to resolve this problem. The answer is to increment "**countIssued**" outside this branch condition. This variable must be very specifically incremented only when there is a relevant property(issued /modified) irrespective of the object. 

In your test case, make **dct:issued     ""** for the dataset, <http://projekt-opal.de/dataset/_mcloudde_standortederdauerzhlstellenradverkehrindsseldorf> and run the JunitTest. You will still get 5 stars. I hope this helps.

Reply:
Nice man, I will look into that, Thanks.